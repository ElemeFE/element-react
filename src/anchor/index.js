import AnchorSmooth from './AnchorSmooth';
import GoTop from './GoTop';

export {AnchorSmooth,GoTop};