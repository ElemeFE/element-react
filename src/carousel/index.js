import Carousel from './Carousel';
import CarouselItem from './CarouselItem';

Carousel.Item = CarouselItem;

export default Carousel;
