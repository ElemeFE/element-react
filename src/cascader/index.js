import Cascader from './Cascader';

export default Cascader;
