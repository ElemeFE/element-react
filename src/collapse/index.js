import Collapse from './Collapse';
import CollapseItem from './CollapseItem';

Collapse.Item = CollapseItem;

export default Collapse;
