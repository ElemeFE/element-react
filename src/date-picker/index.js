/*
revision version: 
  48996c6658300b7f9f01b9b4011a531f17e30b75
*/
export { default as TimeSelect } from './TimeSelect'
export { default as TimePicker } from './TimePicker'
export { default as TimeRangePicker } from './TimeRangePicker'
export { default as DatePicker } from './DatePicker'
export { default as DateRangePicker } from './DateRangePicker'