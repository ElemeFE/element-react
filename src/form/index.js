import Form from './Form';
import FormItem from './FormItem';

Form.Item = FormItem;

export default Form;
