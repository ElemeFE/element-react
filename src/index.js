export { default as i18n } from './locale';

export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Card } from './card';
export { default as Layout } from './layout';
export { default as Loading } from './loading';
export { default as Message } from './message';
export { default as MessageBox } from './message-box';
export { default as Notification } from './notification';
export { default as Radio } from './radio';
export { default as Dialog } from './dialog';
export { default as Rate } from './rate';
export { default as Progress } from './progress';
export { default as Badge } from './badge';
export { default as Tabs } from './tabs';
export { default as Tree } from './tree';
export { default as Input } from './input';
export { default as Icon } from './icon';
export { default as Menu } from './menu';
export { default as Steps } from './steps';
export { default as Breadcrumb } from './breadcrumb';
export { default as Tooltip } from './tooltip';
export { default as InputNumber } from './input-number';
export { default as Checkbox } from './checkbox';
export { default as Slider } from './slider';
export { default as Table } from './table'
export { default as Switch } from './switch';
export { default as Form } from './form';
export { default as Upload } from './upload';
export { default as Tag } from './tag';
export { default as Select } from './select';
export { default as Dropdown } from './dropdown';
export { default as Popover } from './popover';
export { default as Pagination } from './pagination';
export { default as AutoComplete } from './auto-complete';
export { TimeSelect, TimePicker, TimeRangePicker, DatePicker, DateRangePicker } from './date-picker';
export { default as Carousel } from './carousel';
export { default as Collapse } from './collapse';
export { default as ColorPicker } from './color-picker';
export { default as Cascader } from './cascader';
export { default as Transfer } from './transfer';
