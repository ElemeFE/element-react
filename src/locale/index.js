import format from './format';
import defaultLang from './lang/en';

let _lang = defaultLang

function use(lang) {
  _lang = lang;
}

function t(path, options) {
  const array = path.split('.');
    let current = _lang;

    for (var i = 0, j = array.length; i < j; i++) {
      var property = array[i];
      var value = current[property];
      if (i === j - 1) return format(value, options);
      if (!value) return '';
      current = value;
    }
    return '';
}

export default {
  use,
  t
}
