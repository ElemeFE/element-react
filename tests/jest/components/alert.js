import React from 'react';
import renderer from 'react-test-renderer';

import { Alert } from '../../../src';

test('Basic usage', () => {
  const component = renderer.create(
    <div>
      <Alert title="success alert" type="success" />
      <Alert title="info alert" type="info" />
      <Alert title="warning alert" type="warning" />
      <Alert title="error alert" type="error" />
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Customizable close button', () => {
  const component = renderer.create(
    <div>
      <Alert title="unclosable alert" type="success" closable={false} />
      <Alert title="customized close-text" type="info" closeText="Gotcha" />
      <Alert title="alert with callback" type="warning" onClose={() => alert('Hello World!')}/>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('With icon', () => {
  const component = renderer.create(
    <div>
      <Alert title="success alert" type="success" showIcon={true} />
      <Alert title="info alert" type="info" showIcon={true} />
      <Alert title="warning alert" type="warning" showIcon={true} />
      <Alert title="error alert" type="error" showIcon={true} />
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('With description', () => {
  const component = renderer.create(
    <Alert title="with description" type="success" description="This is a description." />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('With icon and description', () => {
  const component = renderer.create(
    <div>
      <Alert title="success alert" type="success" description="more text description" showIcon={true} />
      <Alert title="info alert" type="info" description="more text description" showIcon={true} />
      <Alert title="warning alert" type="warning" description="more text description" showIcon={true} />
      <Alert title="error alert" type="error" description="more text description" showIcon={true} />
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
