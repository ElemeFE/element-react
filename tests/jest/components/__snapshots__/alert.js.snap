// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic usage 1`] = `
<div>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--success"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          success alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--info"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          info alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--warning"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          warning alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--error"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          error alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
</div>
`;

exports[`Customizable close button 1`] = `
<div>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--success"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          unclosable alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--info"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          customized close-text
        </span>
        <i
          className="el-alert__closebtn is-customed"
          onClick={[Function]}
          style={Object {}}
        >
          Gotcha
        </i>
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--warning"
      style={Object {}}
    >
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          alert with callback
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
</div>
`;

exports[`With description 1`] = `
<span
  className={undefined}
  style={undefined}
>
  <div
    className="el-alert el-alert--success"
    style={Object {}}
  >
    <div
      className="el-alert__content"
    >
      <span
        className="el-alert__title is-bold"
      >
        with description
      </span>
      <p
        className="el-alert__description"
      >
        This is a description.
      </p>
      <i
        className="el-alert__closebtn el-icon-close"
        onClick={[Function]}
        style={Object {}}
      />
    </div>
  </div>
</span>
`;

exports[`With icon 1`] = `
<div>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--success"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-circle-check"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          success alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--info"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-information"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          info alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--warning"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-warning"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          warning alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--error"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-circle-cross"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title"
        >
          error alert
        </span>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
</div>
`;

exports[`With icon and description 1`] = `
<div>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--success"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-circle-check is-big"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title is-bold"
        >
          success alert
        </span>
        <p
          className="el-alert__description"
        >
          more text description
        </p>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--info"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-information is-big"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title is-bold"
        >
          info alert
        </span>
        <p
          className="el-alert__description"
        >
          more text description
        </p>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--warning"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-warning is-big"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title is-bold"
        >
          warning alert
        </span>
        <p
          className="el-alert__description"
        >
          more text description
        </p>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
  <span
    className={undefined}
    style={undefined}
  >
    <div
      className="el-alert el-alert--error"
      style={Object {}}
    >
      <i
        className="el-alert__icon el-icon-circle-cross is-big"
      />
      <div
        className="el-alert__content"
      >
        <span
          className="el-alert__title is-bold"
        >
          error alert
        </span>
        <p
          className="el-alert__description"
        >
          more text description
        </p>
        <i
          className="el-alert__closebtn el-icon-close"
          onClick={[Function]}
          style={Object {}}
        />
      </div>
    </div>
  </span>
</div>
`;
